Flagsmith.Client.Test/HttpMocker.cs

```diff
@@ -1,18 +1,50 @@
-﻿using Moq;
-using System;
+﻿using System;
+using System.Collections.Concurrent;
 using System.Collections.Generic;
-using System.Linq;
+using System.IO;
+using System.Net;
 using System.Net.Http;
-using System.Text;
 using System.Threading;
 using System.Threading.Tasks;
 using System.Web;
-using System.Net;
+using Moq;
 
 namespace Flagsmith.FlagsmithClientTest
 {
     internal static class HttpMocker
     {
+        public static ConcurrentBag<string> PayloadsSubmitted { get; set; } = new ConcurrentBag<string>();
+
+        public static Mock<HttpClient> MockHttpResponse(HttpStatusCode statusCode, string content, bool trackPayloads)
+        {
+            var payloadTracker = new Action<IInvocation>(item =>
+            {
+                if (trackPayloads)
+                {
+                    var payload = new StreamReader(((item.Arguments[0] as dynamic).Content as StringContent).ReadAsStream()).ReadToEnd();
+
+                    PayloadsSubmitted.Add(payload);
+                }
+            });
+
+            var httpClientMock = new Mock<HttpClient>();
+            httpClientMock.Setup(x => x.SendAsync(It.IsAny<HttpRequestMessage>(), It.IsAny<CancellationToken>()))
+                .Returns(() =>
+                {
+                    HttpResponseMessage result = new()
+                    {
+                        StatusCode = statusCode,
+                    };
+                    if (content != null)
```

Comment:
This is necessary since the `MockHttpResponse` used by other methods is not thread-safe (since `StringContent` is not thread-safe).

This method thus does not keep returning _the same response_ but a new response / set of content on each invocation.